MUXSDKKaltura/Podfile

```diff
@@ -20,14 +20,64 @@ target 'MUXSDKKalturaTv' do
   use_frameworks!
   
   # Pods for MUXSDKKalturaTv
-  pod 'Mux-Stats-Core', '~>3.11'
+  pod 'MUXSDKStatsShared'
   pod 'PlayKit'
 end
 
 post_install do |installer|
   installer.pods_project.targets.each do |target|
+    
     target.build_configurations.each do |config|
       config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '9.0'
     end
+    
+    if target.name == "Pods-MUXSDKKaltura"
```

Comment:
whoa what is going on here lol. I think I recall you saying Cocoapods overwrites these paths so I'm assuming this is the workaround, but I think I need a deeper explanation 😅

Reply:
Yes of course it was complicated for me to get here haha 
Essentially there is this file that gets created when you run pod install that it's inside the target support files and contains all the build settings according to cocoapods dependencies. Like it's created by cocoapods. In these specific build settings I mention here which are the search paths and the linker flags we have a setting that's $(inherited) which takes whatever this file indicates. If I remove that then it works but it shows a lot of ugly warnings in pod install and if more dependencies are added it could conflict with the new setting required. 

So what this post install hook is doing is taking that file, converting the build settings to a dictionary and modifying the values for those build settings by removing the MUXSDKStatsShared references. The final product is something like this: 
<img width="832" alt="Screen Shot 2022-05-27 at 10 40 26" src="https://user-images.githubusercontent.com/40036667/170722800-985f7091-f48d-4aac-a4f6-f61c0158a7e4.png">

If we don't do this then when we import the sdk into an app like the DemoApp it builds fine but then there's a runtime issue cause when it calls import kaltura it searches for all the linker flags and paths and can't find the dependency of MUXSDKStatsShared.

I like this solution because this post hook script still takes all the other settings so if we update anything it will still contain new settings, it's only part of our internal process and it runs only on pod install. The reason we have to do this is because this is not the classic use case of cocoapods where we would just add the pod as a dependency in the podspec and then the final user would get that dependency, but instead we're actually adding the shared components sdk files as part of the kaltura framework. Because this is somewhat different then cocoapods is kind of rigid regarding changes around the settings they overwrite.